Bottom: 31085ead33a6bfe2a35fe6f724b8d359d6e7ddb2
Top:    fd26fd869b6e2f7959e13084fbe208e426819461
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 19:11:27 +0000

Fix cylindrical surface orientation


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 3f32ee9..13abf64 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -579,10 +579,9 @@ object3d_from_contours (POLYAREA *contours,
 
               /* NOTE: Surface orientation is only fixed up during emission if we flag the need here..
                *       cylindrical surface orientation is always pointing outward from its axis, so
-               *       orientation reversed is used for holes
+               *       orientation reversed is used for holes and convave contour segments.
                */
-              /* XXX: DEPENDS ON INSIDE / OUTSIDE CORNER!! - THIS IS NOT EXACTLY CORRECT!.. NEEDS TO TAKE ACOUNT FOR cw / ccw WHEN DEALING WITH ARC CONTOUR SEGMENTS */
-              if ((ct->Flags.orient == PLF_INV) != extrude_inverted)
+              if (cw != extrude_inverted)
                 face3d_set_surface_orientation_reversed (faces[i]);
 
               face3d_set_normal (faces[i], 1., 0., 0.);  /* A normal to the axis direction */
